common/persistence/nosql/nosql_history_store.go

```diff
@@ -299,7 +299,65 @@ func (h *nosqlHistoryStore) ForkHistoryBranch(
 	return resp, nil
 }
 
-// DeleteHistoryBranch removes a branch
+// DeleteHistoryBranch removes a branch. This is responsible for:
+//
+// - Deleting entries from the history_node table
+// - Deleting the tree entry in the history_tree table
+// - Handling the problem of garbage-collecting branches which may mutually rely on history
```

Comment:
```suggestion
//   - Deleting entries from the history_node table
//   - Deleting the tree entry in the history_tree table
//   - Handling the problem of garbage-collecting branches which may mutually rely on history
```
formatting for godoc, as currently it's like this:

<img width="823" alt="Screenshot 2025-03-19 at 14 11 51" src="https://github.com/user-attachments/assets/de324cb2-a685-4243-a963-d2cc7cefd559" />
though the bullets in the section immediately below it are correct.